keps/sig-apps/85-Graduate-PDB-to-Stable/README.md

```diff
@@ -81,6 +84,20 @@ long term.
 
 ### Implementation Details/Notes/Constraints
 
+#### Allow promotion of Eviction to policy/v1 without breaking pods/eviction support for policy/v1beta1
+
+Eviction is part of policy/v1beta1, but is in a different spot than PDBs since
+it is a subresource of the v1 Pod API. Luckily, the endpoint only supports 
+Create and returns v1 Status, so it should be possible to let the current
+endpoint accept both policy/v1 and policy/v1beta1 Evictions.
+
+The current apiserver decoding stack does allow support for multiple versions
+on a single endpoint, and discovery does not support advertising more than one
+accepted version. If we can add support for accepting multiple versions for a
+subresource endpoint, we can add support for policy/v1 and then phase out
+policy/v1beta1 over a long period of time. There is a discussion about this
+[here](https://github.com/kubernetes/kubernetes/pull/99290#discussion_r582003770).
```

Comment:
Suggest more declarative "this is what we will do" statements:

> Eviction is part of policy/v1beta1, but because it is a subresource of the v1 Pod API,
> support for accepting policy/v1beta1 requests should not be dropped.
> 
> Luckily, the endpoint only supports Create and returns v1 Status,
> so it is possible to let the current endpoint accept both policy/v1 and policy/v1beta1 Evictions.
> 
> The following changes will be made:
> * The decoding stack will be adjusted to allow a REST handler to accept multiple GroupVersionKinds
> * The Eviction subresource handler will accept policy/v1 and policy/v1beta1 Eviction objects
> * Integration tests will be added to ensure:
>   * Get requests continue to be unsupported for this endpoint
>   * Patch requests continue to be unsupported for this endpoint
>   * Create requests continue to accept policy/v1 and policy/v1beta1 requests and return Status objects